[API v4]: create a documentation page from external source

phiz71 opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-5467
https://gravitee.atlassian.net/browse/APIM-5468

## Description


https://github.com/user-attachments/assets/b9b0c1c2-32e3-4dee-b4b9-b8124532c3bd



<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-emztjwmsss.chromatic.com)
<!-- Storybook placeholder end -->

phiz71 committed:
feat(console): create a documentation page from external source

https://gravitee.atlassian.net/browse/APIM-5467
https://gravitee.atlassian.net/browse/APIM-5468

phiz71 commented on gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.ts:
```diff
@@ -172,19 +186,50 @@ export class DocumentationNewPageComponent implements OnInit {
+      .pipe(
+        distinctUntilChanged(),
+        tap((_) => {
+          this.selectedFetcherSchema = undefined;
+          this.sourceConfiguration = undefined;
+        }),
+        debounceTime(500),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe((value) => {
+        this.sourceConfiguration = new FormControl<unknown>({}, [Validators.required]);
+        this.selectedFetcherSchema = this.fetchers.find((f) => f.id === value)?.schema;
```
Comment: Clear variable, wait, and re-init is a hack to force re-rendering of GioJsonSchemaForm.

jourdiw replied:
Might as well add this comment as well to the code?

illichso commented on gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.ts:
```diff
@@ -172,19 +186,50 @@ export class DocumentationNewPageComponent implements OnInit {
+      .subscribe((value) => {
+        this.sourceConfiguration = new FormControl<unknown>({}, [Validators.required]);
+        this.selectedFetcherSchema = this.fetchers.find((f) => f.id === value)?.schema;
+      });
```
Do we need to add error?
Something like
`    error: (err) => {
      console.error('Error handling source value changes:', err);
    },`

jourdiw replied:
I don't think so because there should never be an error on this. And if there is an error, the form will simply not be set.

illichso commented on gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.ts:
```diff
@@ -217,22 +262,24 @@ export class DocumentationNewPageComponent implements OnInit {
     });
   }
 
+  selectFetcherSchema(schema: GioJsonSchema) {
```
Looks like this function is not used anywhere.
Do we need it?

phiz71 replied:
You're right. I forgot to remove this function.

jourdiw commented on gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.html:
```diff
@@ -59,31 +57,59 @@
+          <ng-template matStepLabel>
+            @switch (form.value.sourceType) {
+              @case ('FILL') {
+                {{ 'Add content' }}
```
I believe that the moustaches + quotation marks can be removed for each of the cases:
```suggestion
                Add content
```

phiz71 committed:
feat: bump fetchers

To add the improvements about fetch Cron Expression

https://gravitee.atlassian.net/browse/APIM-5467
https://gravitee.atlassian.net/browse/APIM-5468

jourdiw commented on gravitee-apim-console-webui/src/management/api/documentation-v4/components/documentation-new-page/documentation-new-page.component.spec.ts:
What do you think about adding a test like the bug that you found before? --> User changes source selection and only the specified values of the chosen source are kept.
